c7n/filters/core.py

```diff
@@ -760,3 +760,15 @@ def get_resource_value(self, resource):
         if capture is None:  # regex didn't capture anything
             return None
         return capture.group(1)
+
+
+class StateTransitionFilter(Filter):
```

Comment:
we'll also want to refactor other uses of the state transition filter here too, from a cursory glance it appears that batch, opsworks, ec2, and sagemaker all have their own internal implementations of the StateTransitionFilter. For this PR i'd argue the larger refactor is out of scope atm but that should be an immediate action item after this.

Reply:
opened a separate issue re this here: https://github.com/cloud-custodian/cloud-custodian/issues/4165